### PR TITLE
[SPARK-22778][Kubernetes] Added the missing service metadata for KubernetesClusterManager

### PR DESCRIPTION
**Pull request by liyinan926**
## What changes were proposed in this pull request?

This PR added the missing service metadata for `KubernetesClusterManager`. Without the metadata, the service loader couldn't load `KubernetesClusterManager`, and caused the driver to fail to create a `ExternalClusterManager`, as being reported in SPARK-22778. The PR also changed the `k8s:` prefix used to `k8s://`, which is what existing Spark on k8s users are familiar and used to. 

## How was this patch tested?

Manual testing verified that the fix resolved the issue in SPARK-22778.

/cc @vanzin @felixcheung @jiangxb1987

### CODE REVIEW

**Comment by foxish** [comment]
LGTM.
Will independently verify the fix as well. Thanks @mccheah and @liyinan926 for debugging.

**Comment by liyinan926** [comment]
@vanzin @felixcheung @jiangxb1987 Can one of you help take a look and merge? This is fixing a blocker issue for Kubernetes mode. Thanks!

**Comment by vanzin** [comment]
One of the tests failures looks legitimate...

**Comment by liyinan926** [comment]
@vanzin Yes, fixed in https://github.com/apache/spark/pull/19972/commits/c91c9a606bb00ea57c50fe27081273e0d5a8bcee.

**Comment by felixcheung** [comment]
LGTM

File: resource-managers/kubernetes/core/src/main/resources/META-INF/services/org.apache.spark.scheduler.ExternalClusterManager [hunk 1]
```diff
@@ -0,0 +1 @@
+org.apache.spark.scheduler.cluster.k8s.KubernetesClusterManager
```
Comment (jiangxb1987) [inline resource-managers/kubernetes/core/src/main/resources/META-INF/services/org.apache.spark.scheduler.ExternalClusterManager]: What's this line for?

Comment (jiangxb1987) [inline resource-managers/kubernetes/core/src/main/resources/META-INF/services/org.apache.spark.scheduler.ExternalClusterManager]: nit: add an extra empty line below this

**Review by jiangxb1987 — COMMENTED** [review]
LGTM only one nit

File: resource-managers/kubernetes/core/src/main/resources/META-INF/services/org.apache.spark.scheduler.ExternalClusterManager (hunk 1 above)
Comment (liyinan926) [reply resource-managers/kubernetes/core/src/main/resources/META-INF/services/org.apache.spark.scheduler.ExternalClusterManager]: Done.

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**